### PR TITLE
Rip out drake/matlab/systems from the drake/systems directory

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
A fourth portion of issue #3253.

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
Finally passing pre-merge CI!

+@david-german-tri for review and local testing please.  As with the prior one, I suggest using github review of commit-by-commit, due to the 930 files changed making reviewable explode.

@drake-jenkins-bot linux-gcc-ninja-experimental please
@drake-jenkins-bot linux-gcc-ninja-experimental-matlab please
@drake-jenkins-bot mac-clang-experimental please
@drake-jenkins-bot mac-clang-experimental-matlab please
@drake-jenkins-bot mac-clang-ninja-experimental please
@drake-jenkins-bot mac-clang-ninja-experimental-matlab please

**Comment by jwnimmer-tri** [comment]
@david-german-tri How goes?

**Comment by david-german-tri** [comment]
Bad news - you have merge conflicts.

Other bad news - I still can't make this work locally, for reasons that are probably unrelated.

**Comment by david-german-tri** [comment]
As discussed f2f, please take a grep through Sphinx and make sure this doesn't affect any documented examples.  Otherwise, :lgtm: pending merge conflict resolution, CI, and me figuring out how to get this to run locally.

---

Review status: 0 of 627 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3861#-:-KUx82ZkPVwmNuw6qX64:bmq8okp)*

File: drake/matlab/systems/CPPLINT.cfg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# TODO(#2274) Fix this.
+filter=-runtime/references
+
+# TODO(#2271) Fix this.
+# Once matlab code is evicted, it should be a simple as removing this line.
```
Comment (jwnimmer-tri) [inline drake/matlab/systems/CPPLINT.cfg]: Right.  I have a few cpplint tune-ups on deck behind this PR (and the issue numbers will track that reasonably well).

**Comment by jwnimmer-tri** [comment]
Tagging DNM pending David's ability to test this locally.

**Comment by jwnimmer-tri** [comment]
I have rebased around the merge conflicts (just the new "trajectories" unit test from #3856), so let's fire off the tests again:

@drake-jenkins-bot linux-gcc-ninja-experimental please
@drake-jenkins-bot linux-gcc-ninja-experimental-matlab please
@drake-jenkins-bot mac-clang-experimental please
@drake-jenkins-bot mac-clang-experimental-matlab please
@drake-jenkins-bot mac-clang-ninja-experimental please
@drake-jenkins-bot mac-clang-ninja-experimental-matlab please

**Comment by jwnimmer-tri** [comment]
> ... please take a grep through Sphinx and make sure this doesn't affect any documented examples.

I've browsed through all of the `doc/*` files in my editor (including with search on key terms), and didn't notice any citations that need changing in this PR.

**Comment by jwnimmer-tri** [comment]
Flakes!  Must be almost wintertime.
@drake-jenkins-bot mac-clang-ninja-experimental-matlab please

**Comment by jwnimmer-tri** [comment]
This message well self-destruct in 30 seconds.

@drake-jenkins-bot linux-gcc-ninja-experimental please
@drake-jenkins-bot linux-gcc-ninja-experimental-matlab please
@drake-jenkins-bot mac-clang-experimental please
@drake-jenkins-bot mac-clang-experimental-matlab please
@drake-jenkins-bot mac-clang-ninja-experimental please
@drake-jenkins-bot mac-clang-ninja-experimental-matlab please

**Comment by jwnimmer-tri** [comment]
This post will self-destruct in 30 seconds

@drake-jenkins-bot linux-gcc-ninja-experimental please
@drake-jenkins-bot linux-gcc-ninja-experimental-matlab please
@drake-jenkins-bot mac-clang-experimental please
@drake-jenkins-bot mac-clang-experimental-matlab please
@drake-jenkins-bot mac-clang-ninja-experimental please
@drake-jenkins-bot mac-clang-ninja-experimental-matlab please

**Comment by david-german-tri** [comment]
:lgtm:

---

Review status: 0 of 627 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3861#-:-KV0ILbKZF9PHnkQCNbh:bnfp4nl)*

**Comment by jwnimmer-tri** [comment]
David says local testing passed.  Merging!

**Pull request merged**